flybywiresim/aircraft
# feat(EFB): Simbrief Loadsheet/OFP Navigation

## Summary of Changes


- [x] **Feature:** Navigation through the different Loadsheet/OFP sections in the EFB, improving the reading speed and efficiency.
- [x] It supports these OFP Layouts: [LIDO, AAL, ACA, BAW, DAL, DLH, EIN, EZY, JBU, QFA, THY, BER, KLM]

## Screenshots (if necessary)



https://github.com/user-attachments/assets/d407a8ab-7cfe-47a7-bfdc-5b0d2cd1a61f

![Loadsheet Navigation with EZY OFP Layout](https://github.com/user-attachments/assets/e8cee49f-69b4-43d1-991e-7dd3a851d408)

## References


e.g LIDO Format:
![Screenshot 2024-11-11 181233](https://github.com/user-attachments/assets/f5e53dc3-cf90-4eef-bed3-6b111b6f0837)




## Additional context

OFP Layouts: [UAL-2018, UAL-2012, AFR-2012, AFR-2017] were tested but table formatting gets broken, so i decided to not support them for now. I think the issue is caused by the font scaling feature in the loadsheet.

Discord username (if different from GitHub): itzarz

## Testing instructions


1. Go to Simbrief and generate a flight (make sure to use the Official FBW A20N/A380 Profile).
2. Import the simbrief data through the EFB.
3. Select the **Dispatch** option in the EFB sidebar.
4. Select a desired section and ensure if the feature is working.
5. Compare with Simbrief PDF Sections and its content. 
6. Repeat all the steps with every supported OFP Layout.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo**, **flybywire-aircraft-a380-842 (4K)** or **flybywire-aircraft-a380-842 (8K)** download link at the bottom of the page
